core/src/test/java/org/infinispan/partitionhandling/PartitionHigherTopologyIDTest.java

```diff
@@ -0,0 +1,87 @@
+package org.infinispan.partitionhandling;
+
+import static org.testng.AssertJUnit.assertEquals;
+
+import org.infinispan.distribution.LocalizedCacheTopology;
+import org.infinispan.factories.ComponentRegistry;
+import org.infinispan.test.TestingUtil;
+import org.infinispan.topology.ClusterTopologyManager;
+import org.jgroups.JChannel;
+import org.jgroups.protocols.MERGE3;
+import org.testng.annotations.Test;
+
+/**
+ * Partition tests when one sub group has a higher topology than other sub groups.
+ */
+@Test(groups = "functional", testName = "conflict.impl.MultipleCachesDuringConflictResolutionTest")
+public class PartitionHigherTopologyIDTest extends BasePartitionHandlingTest {
+
+    private final PartitionDescriptor p0;
+    private final PartitionDescriptor p1;
+
+    public PartitionHigherTopologyIDTest() {
+        this.p0 = new PartitionDescriptor(0, 1);
+        this.p1 = new PartitionDescriptor(2, 3);
+        this.partitionHandling = PartitionHandling.ALLOW_READ_WRITES;
+    }
+
+    public void testPartitionWithDifferentTopologyIds() {
+        TestingUtil.waitForNoRebalance(caches());
+
+        log.tracef("split test");
+        splitCluster(p0.getNodes(), p1.getNodes());
+        TestingUtil.waitForNoRebalance(caches());
+
+        ClusterTopologyManager manager = ComponentRegistry.of(cache(0)).getComponent(ClusterTopologyManager.class);
+
+        // We toggle the availability to increase the topology id
+        log.tracef("Forcing available status");
+        manager.forceAvailabilityMode("defaultcache", AvailabilityMode.AVAILABLE);
+
+        TestingUtil.waitForNoRebalance(getPartitionCaches(p0));
+
+        log.tracef("Forcing degraded status");
+        manager.forceAvailabilityMode("defaultcache", AvailabilityMode.DEGRADED_MODE);
+
+        TestingUtil.waitForNoRebalance(getPartitionCaches(p0));
+
+        log.tracef("performMerge");
```

Comment:
Should we verify the topology ids before healing?

Reply:
I can add it.

Reply:
Updated with checks to ensure. It actually wasn't working before :)